roles/infra/tasks/main.yml

```diff
@@ -1,23 +1,54 @@
 ---
 
-- name: Write backend configuration
+- name: Write backend type configuration
   copy:
     content: |
       terraform {
         backend "{{ terraform_backend_type }}" { }
       }
     dest: "{{ terraform_project_path }}/backend.tf"
 
+- name: Write backend configuration options
+  copy:
+    content: "{{ terraform_backend_config | to_json }}"
+    dest: "{{ terraform_project_path }}/backend_config.json"
+
 - name: Provision infrastructure using Terraform
```

Comment:
Not related to this PR, but we should change this task name because we don't always provision here - maybe "Update infrastructure using Terraform" would be better.